scertecd/scertecd.go

```diff
@@ -140,12 +215,45 @@ func (s *Server) Start(ctx context.Context) error {
 	}
 
 	go s.checkAWSPermissionsLoop()
-	for _, dt := range s.dts {
-		go s.renewCertLoop(dt)
-	}
+	go s.renewCertsLoop()
 	return nil
 }
 
+// renewCertsLoop periodically ensures the certificates scertecd is monitoring are
+// valid.
+//
+// If DynamicDomainsKey is set, the list of additional domains is also updated
+// periodically.
+func (s *Server) renewCertsLoop() {
+	for {
+		// Refresh our understanding of any dynamic domains.
+		if s.DynamicDomainsKey != "" {
+			// Set a long timeout, just so we progress eventually even if there
+			// is an issue.
+			ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+			if err := s.updateDynamicDomainList(ctx); err != nil {
+				s.Logf("updateDynamicDomainList error: %v", err)
+			}
+			cancel()
+		}
+
+		// Grab a copy of the list of certs we should do renewal runs for, in da lock.
+		s.mu.Lock()
+		dts := make([]domainAndType, len(s.dts))
```

Comment:
```go
dts := slices.Clone(s.dts)
```
maybe?

Reply:
Yo see, that just sounds like a great idea